lib/tiled/map.rb

```diff
@@ -89,6 +89,28 @@ def pixelheight
       return height * tileheight
     end
 
+    #gets the first object layer with the passed name of `layer_name`
+    #then looks in that layer for the first object that has the name `object_name`
+    #then returns that object
+    def object_by_name(layer_name, object_name)
+      
+      named_layer = @object_groups.reject do | layer |
```

Comment:
It is better to use `object_groups` method instead of instance variable, because it can be uninitialized if map file doesn't have object layers.

`object_group` is a `Tiled::Layers` instance, so you can work with it like with `map.layers`:

```ruby
named_layer = object_groups[layer_name]
```

As a bonus it will cache reference to this layer.